pyproject.toml

```diff
@@ -1,3 +1,58 @@
+[tool.poetry]
+package-mode = false
+[tool.poetry.group.dev.dependencies]
```

Comment:
How did you settle on these groups? I'm not sure what we're doing now is equivalent to what we had before

Reply:
I settled on using three groups:
- Dev  - dependencies exclusive to requirements-dev.lock, explicitly mentioned in the docs
- Deployment - dependencies exclusive to requirements.lock
- No group (main group) - dependencies shared between the two

The intent is that main group loads in both scenarios, while deployment is only included when explicitly specified during `poetry install --with deployment` and dev seemed to have special functionality within Poetry according to the docs, so its only included in dev environments.

Have I misunderstood how this works?

Reply:
Got you - I think it was that missing `--with dev` that was the issue as you already spotted.